### PR TITLE
window-dimensions and zoom crash fixed

### PR DESCRIPTION
**Pull request by zaidnadaff**
fixed issue #1613 partially, 

added window:dimensions as a setting which gives the user to define dimensions.

fixed the zoom in and out logic to prevent the app from crashing.

Had issues with getting setting zoomfactor in emain-tabview.ts. will fix it

### CODE REVIEW

File: emain/emain-window.ts [hunk 1]
```diff
@@ -73,11 +73,32 @@ export class WaveBrowserWindow extends BaseWindow {
...
+            const parsedWidth = parseInt(dimensionWidth);
+            const parsedHeight = parseInt(dimensionHeight);
+
+            if ((!winWidth || winWidth === 0) && !isNaN(parsedWidth)) {
+                winWidth = parsedWidth;
+            }
+            if ((!winHeight || winHeight === 0) && !isNaN(parsedHeight)) {
+                winHeight = parsedHeight;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (esimkowitz) [reply emain/emain-window.ts:100]: @zaidnadaff this is a good suggestion

File: emain/emain-window.ts [hunk 2]
```diff
@@ -73,11 +73,32 @@ export class WaveBrowserWindow extends BaseWindow {
...
+
         if (winWidth == null || winWidth == 0) {
```
Comment (esimkowitz) [inline emain/emain-window.ts:102]: I see you haven't added the zoom factor yet, do you want to incorporate that into this PR or a follow-up? Could you also add the new settings to `config.mdx`?

Comment (zaidnadaff) [reply emain/emain-window.ts:107]: I will incorporate the zoom factor in a follow-up PR.

**Comment by zaidnadaff** [comment]
Sure! 😀

**Pull request merged**